### PR TITLE
Premium Blocks: Handle core/audio block

### LINKED ISSUE #14902: Add upgrade nudge to Audio block for Simple site free plan users
Matching the result in https://github.com/Automattic/jetpack/pull/14735 
Related: https://github.com/Automattic/wp-calypso/issues/33013

### PR DESCRIPTION
**Pull request by retrofox**
This PR makes handle the ` core/audio` block upgrade stuff. For this purpose, it implements a refactoring moving some functions, code, etc. to finally remove completely the `extensions/shared/blocks/cover` folder. Everything is handled by premium blocks.

Fixes #14902

#### Changes proposed in this Pull Request:

* Premium Blocks: handle `core/audio` upgrade block

#### Does this pull request change what data or activity we track or use?

no

#### Testing instructions:

* Get sandboxed
* Apply D46955-code and the changes implemented on this PR https://github.com/Automattic/jetpack/pull/16636
* Test in a Simple site with a free plan
* Add a `core/cover` block
* Perform the testing flow for the core/cover block
  - Try up upload a video dropping a video file in the media placeholder area.
  - Try to replace the background uploading a video via the replace control button

* Add a `core/audio` block
  - Try to upload an audio file - confirm that you get the upgrade banner when it tries to upload it.
image

  - Try to upload an unsupported file - confirm you get a generic error.
image

#### Proposed changelog entry for your changes:

* Premium Blocks: handle `core/audio` upgrade block

### CODE REVIEW

File: extensions/shared/get-allowed-mime-types.js [hunk 1]
```diff
@@ -48,3 +48,35 @@ export function pickFileExtensionsFromMimeTypes( mimeTypesObject ) {
...
 }
+
+/**
+ * Check if the given file is a video.
+ *
+ * @param   {string|object} file - File to check.
+ * @param   {string}        type - File type used to check the file.
+ * @returns {boolean}       True if file type matches with the given type. Otherwise, False.
+ */
+export function checkFileType( file, type ) {
```
Comment (obenland) [inline extensions/shared/get-allowed-mime-types.js:59]: How would you feel about calling it `isVideoFile`?

Comment (retrofox) [reply extensions/shared/get-allowed-mime-types.js:59]: `isVideoFile` was the name before to this refactoring when we needed to check only video files. Now we are checking different types:

```es6
if ( checkFileType( file, 'video' ) {
  // is a video!
} else if ( checkFileType( file, 'audio' ) {
  // is an audio!
} else {
  // who knows
}
```

I did this refactoring just because we need to check if the file dropped in the `core/audio` block is an audio file.

**Comment by obenland** [comment]
I missed that, my bad. How would you feel about `isFileOfType`?

**Comment by retrofox** [comment]
> I missed that, my bad. How would you feel about `isFileOfType`?

Tackled [here](https://github.com/Automattic/jetpack/pull/16717/commits/06e2baa4d43711cd7f5ea9bc86310782fbd538fa)

**Comment by retrofox** [comment]
I've [added replace media flow](https://github.com/Automattic/jetpack/pull/16717/commits/dc0415045f2c5d700337ade1172f386200dd43ba) for `core/audio` block, because it's possible to try to upload an audio file replacing by the `Replace` control button.

**before**
image

**after**
image

File: extensions/shared/plan-utils.js [hunk 2]
```diff
@@ -101,6 +122,8 @@ export function isUpgradable( name ) {
...
  * @returns {boolean} True is the block is usable with a Free plan. Otherwise, False.
  */
-export const isStillUsableWithFreePlan = ( name ) => [
-	'core/cover',
-].includes( name );
+export const isStillUsableWithFreePlan = name =>
+	map( usableBlockWithFreePlan, 'name' ).includes( name );
+
+export const getUsableBlockProps = blockName =>
+	head( filter( usableBlockWithFreePlan, ( { name } ) => name === blockName ) );
```
Comment (obenland) [inline extensions/shared/plan-utils.js:129]: Would this work?

```suggestion
	pickBy( usableBlockWithFreePlan, ( { name } ) => name === blockName );
```

**Review by cpapazoglou — COMMENTED** [review]
Tested by test instructions and all seems working as advertised. Just noticed that the upgrade banner in the inspector doesn't appear in the correct place for the `core/cover`.



I have only found a corner case which already exists in production so I am going to open a separate issue to wrangle it (https://github.com/Automattic/wp-calypso/issues/44720). It goes as follows:
- a user has a premium plan
- uploads a ton of videos
- downgrades personal or to FREE (cancel subscription)
- if the user uses /wp-admin then they can use these videos already uploaded

**Comment by retrofox** [comment]
> Tested by test instructions and all seems working as advertised. Just noticed that the upgrade banner in the inspector doesn't appear in the correct place for the `core/cover`.

yeah, it's a known issue. We already talked about this and decided to remove the hack.

> I have only found a corner case which already exists in production so I am going to open a separate issue to wrangle it (Automattic/wp-calypso#44720).

Yup, I think it's out of the scope of this PR.

Thanks for reviewing

**Comment by retrofox** [comment]
> Tested by test instructions and all seems working as advertised. Just noticed that the upgrade banner in the inspector doesn't appear in the correct place for the `core/cover`.

maybe we can remove the block styles if the block needs an upgrade, or simply remove the banner for dual-mode blocks.

**Review by obenland — APPROVED** [review]
